File: euler1.js
function sum_limit_step(limit,step) {
    var result = 0
    for (i = step; i < limit; i += step) {
        result += i;
    }
    return result;
}

function euler1(x, y, limit) {
    return (sum_limit_step(limit,x) +
            sum_limit_step(limit,y) -
            sum_limit_step(limit, (x*y)))
}

WScript.echo(euler1(3,5,1000));
//WScript.quit();

File: euler1.scala
object euler1 {
	def main (args: Array[String]){
		println(e1sum(3,5,1000))
	}
	
	def sumStepRange (x: Int, step: Int) = { 
		val l = List.range(step, x, step)
		l.foldLeft(0)(_ + _)}
		
	def e1sum (x: Int, y: Int, limit: Int) = {
		sumStepRange(limit, x) +
		sumStepRange(limit, y) -
		sumStepRange(limit, (x*y))
	}
		
	


}


